### PR TITLE
Fully working recipe page

### PR DESCRIPTION
**Pull request by nhilikeknee**
Restructured recipe page code to work with latest search implementation, restyled settings page, recipe cards, and category cards. The router functionality is temporary.

### CODE REVIEW

File: source/scripts/Router.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Router.js
+export class Router {
+    static routes = {};
+  
+    constructor(homeFunc) {
+      this['home'] = homeFunc;
```
Comment (kjl001) [inline source/scripts/Router.js:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422335)
```suggestion
      this["home"] = homeFunc;
```

File: source/scripts/script.js [hunk 2]
```diff
@@ -1,7 +1,20 @@
 // Import API from file
 import { fetchRecipes } from "./api_script.js";
+import { Router } from './Router.js';
```
Comment (kjl001) [inline source/scripts/script.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422335)
```suggestion
import { Router } from "./Router.js";
```

File: source/scripts/main.js [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+/**
+ * When user clicks on bookMark icon, it saves recipe's title and ID to localStorage
+ * If the recipe already saved, clicking it again will remove it from localStorage.
+ * The data being stored in 'bookmark'
+ */
+function setBookMark() {
+    // check local storage for bookmark
+    let bookmarkList = JSON.parse(localStorage.getItem("bookmark"));
+    if (bookmarkList == null)
+        bookmarkList = {};
```
Comment (kjl001) [inline source/scripts/main.js:179]: Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422335)
```suggestion
        {bookmarkList = {};}
```

File: source/scripts/main.js [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+function hideCategoryCards() {
+    const categoryCards = document.getElementById("category-wrapper");
+    categoryCards.style.visibility = "hidden";
+}
+
+function updateSettings() {
+    const dietaryRestrictionList = [];
+    // Get all the inputs under the div
+    const dietaryContainerElements = document.getElementById('dietary-container').elements;
```
Comment (kjl001) [inline source/scripts/main.js:126]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422335)
```suggestion
    const dietaryContainerElements = document.getElementById("dietary-container").elements;
```

File: source/scripts/script.js [hunk 5]
```diff
@@ -87,193 +98,45 @@ function search() {
...
-function bindRecipeCard(recipeCard) {
-    recipeCard.addEventListener('click', event => {
-        console.log("Recipe card has been clicked");
-        const recipePageContainer = document.querySelector("#recipe-page-container");
-        hideHome();
-        hideRecipeCards();
-        recipePageContainer.classList.remove("hidden");
+function bindRecipeCard(recipeCard, pageName) {
+    recipeCard.addEventListener('click', e => {
+        if (e.path[0].nodeName == 'A') return;
```
Comment (kjl001) [inline source/scripts/script.js:127]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422335)
```suggestion
        if (e.path[0].nodeName == "A") return;
```

File: source/scripts/main.js [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+        if (getIntolerancesRestrictions && getIntolerancesRestrictions.includes(intoleranceRestriction.value)) {
+            intoleranceRestriction.checked = true;
+        }
+    }
+}
+
+function clearCheckBoxes() {
+    let checkboxes = document.querySelector("#recipe-page-container > recipe-page")
+        .shadowRoot.querySelectorAll("#ingredients-list > ul > ol > input");
+    checkboxes.forEach(e => e.checked = false);
```
Comment (kjl001) [inline source/scripts/main.js:54]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422335)
```suggestion
    checkboxes.forEach((e) => e.checked = false);
```

File: source/scripts/script.js [hunk 7]
```diff
@@ -87,193 +98,45 @@ function search() {
...
+        router.navigate(pageName);
     });
 }
 
+function bindPopstate() {
+    window.addEventListener("popstate", (event) => {
+        if (event.state != null)
+            router.navigate(event.state, true);
```
Comment (kjl001) [inline source/scripts/script.js:135]: Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422335)
```suggestion
            {router.navigate(event.state, true);}
```

**Comment by nhilikeknee** [comment]
See #29 for bug about failed search if dietary/intolerances are empty.

**Comment by Dsuzuki126** [comment]
i think its good just a few things as we keep working on it
1) when I click the back arrow from a recipe page i thought I would go back to see the search results from clicking a category card but I just straight back so may be an issue if the user expects to go back to the search results
refreshing the recipe page also brings you back to the home page
2) refreshing the recipe page also brings you back to the home page instead of just refreshing the page

**Pull request merged**